feature/floor-map/src/main/java/io/github/droidkaigi/confsched2023/floormap/FloorMapScreen.kt

```diff
@@ -115,42 +120,108 @@ private fun FloorMapScreen(
-                    FloorMapSideEventList(
-                        uiState = uiState.floorMapSideEventListUiState,
+                // FIXME The LargeScreen decision is the magic coat. Any good ideas?
+                val density = LocalDensity.current.density
+                if (density > 2.5f && maxWidth >= 400.dp) {
+                    LargeScreenContent(
+                        uiState = uiState,
                         onSideEventClick = onSideEventClick,
+                    )
+                } else {
+                    MobileContent(
+                        uiState = uiState,
+                        onSideEventClick = onSideEventClick,
+                    )
+                    FloorLevelSwitcher(
+                        selectingFloorLevel = uiState.floorLevel,
+                        onClickFloorLevelSwitcher = onClickFloorLevelSwitcher,
                         modifier = Modifier
-                            .weight(1f)
-                            .padding(bottom = 56.dp)
-                            .fadingEdge(
-                                Brush.verticalGradient(
-                                    0.85f to Color.Black,
-                                    1f to Color.Transparent,
-                                ),
-                            ),
+                            .align(Alignment.BottomCenter)
+                            .padding(bottom = 24.dp),
                     )
                 }
-                FloorLevelSwitcher(
-                    selectingFloorLevel = uiState.floorLevel,
-                    onClickFloorLevelSwitcher = onClickFloorLevelSwitcher,
-                    modifier = Modifier
-                        .align(Alignment.BottomCenter)
-                        .padding(bottom = 24.dp),
-                )
             }
-        },
+        }
     )
 }
 
+@Composable
+private fun MobileContent(
+    uiState: FloorMapScreenUiState,
+    onSideEventClick: (url: String) -> Unit,
+) {
+    Column(
+        modifier = Modifier.fillMaxSize()
+    ) {
+        FloorMap(uiState = uiState.floorMapUiState)
+        FloorMapSideEventList(
+            uiState = uiState.floorMapSideEventListUiState,
+            onSideEventClick = onSideEventClick,
+            modifier = Modifier
+                .weight(1f)
+                .padding(bottom = 56.dp)
+                .fadingEdge(
+                    Brush.verticalGradient(
+                        0.85f to Color.Black,
+                        1f to Color.Transparent,
+                    ),
+                ),
+        )
+    }
+}
+
+@Composable
+private fun LargeScreenContent(
+    uiState: FloorMapScreenUiState,
+    onSideEventClick: (url: String) -> Unit,
+) {
+    Row(
+        Modifier
+            .fillMaxSize(),
+    ) {
+        Column(
+            modifier = Modifier.weight(0.8f)
+        ) {
+            FloorMap(uiState = FloorMapUiState.of(Basement))
+            FloorMapSideEventList(
```

Comment:
[Help wanted] FloorMapSideEventListUiState is depends on floorLevelStateFlow.
It looks like I should separate the FloorMapScreenViewModel to handle this.
any suggestions?